### PR TITLE
Unexpected return/break should raise LocalJumpError

### PR DESCRIPTION
**Pull request by takaram**
Orphaned `Proc` objects will throw `LocalJumpError` on return or break.
See https://ruby-doc.org/3.2.2/Proc.html#class-Proc-label-Orphaned+Proc

Some related bugs are fixed too.

### CODE REVIEW

File: lib/opal/nodes/closure.rb [hunk 1]
```diff
@@ -223,11 +225,13 @@ def compile_catcher
             end
             line "throw $e;"
           end
-          line "}"
+          line "} finally {", *catchers_without_eval_return.map { |type| "$t_#{type}.is_orphan = true;" }, "}"
```
Comment (hmdne) [inline lib/opal/nodes/closure.rb:228]: Can we skip the finally clause if `catchers_without_eval_return` is empty?

Comment (takaram) [reply lib/opal/nodes/closure.rb]: Yes, I changed the code.

**Review by elia — APPROVED** [review]
This is great! Not the easiest part to tackle 🙌 🙌 🙌

**Comment by hmdne** [comment]
This PR caused a significant failure at some codebase I'm working on (namely - https://github.com/plurimath-js/plurimath-js ). I will give more info when I get to debug it.

**Pull request merged**